gdpr/impl_test.go

```diff
@@ -710,27 +785,44 @@ func TestAllowActivitiesBidRequests(t *testing.T) {
 		purpose2EnforcePurpose string
 		purpose2EnforceVendors bool
 		bidder                 openrtb_ext.BidderName
+		bidderCoreName         openrtb_ext.BidderName
 		consent                string
 		allowBid               bool
 		passGeo                bool
 		passID                 bool
 		weakVendorEnforcement  bool
+		aliasGVLIDs            map[string]uint16
 	}{
 		{
 			description:            "Bid blocked - p2 enabled, user consents to p2 but not vendor, vendor consents to p2",
 			purpose2EnforcePurpose: config.TCF2FullEnforcement,
 			purpose2EnforceVendors: true,
 			bidder:                 openrtb_ext.BidderPubmatic,
+			bidderCoreName:         openrtb_ext.BidderPubmatic,
 			consent:                purpose2ConsentWithoutVendorConsent,
 			allowBid:               false,
 			passGeo:                false,
 			passID:                 false,
 		},
+		{
+			description:            "Pubmatic Alias VendorGVLID, Bid blocked - p2 enabled, user consents to p2 but not vendor, vendor consents to p2",
+			purpose2EnforcePurpose: config.TCF2FullEnforcement,
+			purpose2EnforceVendors: true,
+			bidder:                 "pubmatic1",
+			bidderCoreName:         openrtb_ext.BidderPubmatic,
+			consent:                purpose2ConsentWithoutVendorConsent,
+			allowBid:               false,
+			passGeo:                false,
+			passID:                 false,
+			aliasGVLIDs:            map[string]uint16{"pubmatic1": 6},
+		},
+
```

Comment:
How about replacing this new test case with these to prove that the alias GVL ID has a direct effect on whether the bid is allowed:
```
{
			description:            "Bid allowed - p2 enabled, user consents to p2 and vendor, alias vendor consents to p2",
			purpose2EnforcePurpose: config.TCF2FullEnforcement,
			purpose2EnforceVendors: true,
			bidder:                 "pubmatic1",
			bidderCoreName:         openrtb_ext.BidderPubmatic,
			consent:                purpose2AndVendorConsent,
			allowBid:               true,
			passGeo:                false,
			passID:                 true,
			aliasGVLIDs:            map[string]uint16{"pubmatic1": 6},
		},
		{
			description:            "Bid blocked - p2 enabled, user consents to p2 and vendor, alias vendor does not consent to p2",
			purpose2EnforcePurpose: config.TCF2FullEnforcement,
			purpose2EnforceVendors: true,
			bidder:                 "pubmatic1",
			bidderCoreName:         openrtb_ext.BidderPubmatic,
			consent:                purpose2AndVendorConsent,
			allowBid:               false,
			passGeo:                false,
			passID:                 false,
			aliasGVLIDs:            map[string]uint16{"pubmatic1": 1},
		},
```

Reply:
Replaced the existing test case with these two test cases.